Created pipeline for ECR and ECS

OusainouJ opened the pull request:
Will add trivy in next commit

HamzahS-101 commented:
Could you remove the ECS stuff at the end please as well as all of the terraform stuff? I'm not sure the what this is for but we can discuss it later. For now we'd like this workflow to only build the docker image and push to ECR (And then add the trivy checks later).

The command you used for the docker build step will pull all of the files into the image including the terraform files. Use this instead please:

docker build -t ${{ env.ECR_REPO }}:latest -f Dockerfile app/

OusainouJ commented:
@HamzahS-101 The reason i added the terraform part was to automate the terraform deployment within the pipeline but i've removed it for now and added the docker build change

HamzahS-101 commented:
Try using these actions instead for configuring the AWS credentials and logging into ECR:

aws-actions/configure-aws-credentials@v4

aws-actions/amazon-ecr-login@v2

With these, you don't have to run a command to sign-in and you don't need to set environment variables.

Also, split the build and push steps.

HamzahS-101 commented:
Looks good. Merging into dev branch.
